lti_consumer/plugin/views.py

```diff
@@ -130,3 +144,84 @@ def get_queryset(self):
     def perform_create(self, serializer):
         lti_configuration = self.request.lti_configuration
         serializer.save(lti_configuration=lti_configuration)
+
+    @action(
+        detail=True,
+        methods=['GET'],
+        renderer_classes=[LineItemResultsRenderer]
+    )
+    def results(self, request, *args, **kwargs):
+        """
+        Return a Result list for an LtiAgsLineItem
+
+        Query Parameters:
+          * user_id (string): String external user id representation. Only
+                one record per unique `user_id` will be returned
+          * limit (integer): The maximum number of records to return. Records are
+                sorted with most recent timestamp first
+
+        Returns:
+          * An array of Result records, formatted by LtiAgsResultSerializer
+                and returned with the media-type for LineItemResultsRenderer
+        """
+        line_item = self.get_object()
+        scores = line_item.scores.all().order_by('-timestamp')
```

Comment:
Haven't done a full test yet, but I think this can be replaced with a simple queryset (given the database constraints implemented in `models.py`).

Reply:
@giovannicimolin I'm not sure what you mean by this. Could you explain?

Reply:
@pcockwell I'm a bit confused on what the for loop below is doing. I don't think it's necessary since only user only has a single Score for each LineItem.

Reply:
This is an outdated view. Here is the new code

https://github.com/open-craft/xblock-lti-consumer/blob/0a98de3921aa65ad507620c253957509ef084d4b/lti_consumer/plugin/views.py#L167-L182